source/extensions/quic_listeners/quiche/envoy_quic_client_stream.cc

```diff
@@ -132,25 +134,42 @@ void EnvoyQuicClientStream::switchStreamBlockState(bool should_block) {
 
 void EnvoyQuicClientStream::OnInitialHeadersComplete(bool fin, size_t frame_len,
                                                      const quic::QuicHeaderList& header_list) {
-  if (rst_sent()) {
+  if (read_side_closed()) {
     return;
   }
   quic::QuicSpdyStream::OnInitialHeadersComplete(fin, frame_len, header_list);
   ASSERT(headers_decompressed() && !header_list.empty());
 
-  response_decoder_->decodeHeaders(
-      quicHeadersToEnvoyHeaders<Http::ResponseHeaderMapImpl>(header_list),
-      /*end_stream=*/fin);
+  ENVOY_STREAM_LOG(debug, "Received headers: {}.", *this, header_list.DebugString());
   if (fin) {
     end_stream_decoded_ = true;
   }
+  std::unique_ptr<Http::ResponseHeaderMapImpl> headers =
+      quicHeadersToEnvoyHeaders<Http::ResponseHeaderMapImpl>(header_list);
+  const uint64_t status = Http::Utility::getResponseStatus(*headers);
+  if (Http::CodeUtility::is1xx(status)) {
```

Comment:
So what happens with 1xx headers which are not 100-continue?

For non-QUIC we send them up via decodeHeaders.

Reply:
Oh, I didn't know that. Will fix. Question please: QUIC doens't support 101 response code: https://tools.ietf.org/html/draft-ietf-quic-http-32#section-4.3. What shall we do in that case?

Reply:
oh fair point.  maybe treat it as invalid and fast-fail the response then?

Reply:
Re-wrote 1xx header handling logic here. PTAL!